syft/frameworks/torch/federated/dataset.py

```diff
@@ -13,49 +13,137 @@ class BaseDataset:
     of a .data attribute for inputs and a .targets one for labels. It is to
     be used like the MNIST Dataset object, and is useful to avoid handling
     the two inputs and label tensors separately.
+    
+    Args: 
+        
+        data[list,torch tensors]: the data points
+        targets: Corresponding labels of the data points
+        transform: Function to transform the datapoints 
+            
     """
 
-    def __init__(self, data, targets):
+    def __init__(self, data, targets, transform=None):
+
         self.data = data
         self.targets = targets
 
+        # Perform required transform on the dataset
+        if transform:
+
+            self.data = transform(self.data)
+
     def __len__(self):
+
+        """
+        returns:
```

Comment:
I don't think we need docstring for __len__ and __getitem__ since they are self explanable.